### PR TITLE
Remove the state map for ensembles

### LINKED ISSUE #6736: Remove HAS_DATA and INITIALIZED from `state_map`
`HAS_DATA` means the ensemble has responses loaded, while `INITIALIZED` means the ensemble has parameters. We read this information from the state map currently, but that only holds one state, which means that `HAS_DATA` often replaces `INITIALIZED`. This is a problem, because we would often like to check both, but that is not possible.

One example where we could have data, but not parameters, is cases where `Load results manually` is used. The state_map will then have state: `HAS_DATA´, and we could use it as prior in a manual update, however that is not viable because it does not have parameters.

Instead we can get this information by combining the information in the `ensemble` and the `experiment`, by checking if the expected parameters are present for INITIALIZED and the expected responses are present for HAS_DATA.

To reproduce:
1. Start gui (for example poly.ert)
2. Note that it is not possible to run `Run analysis` due to no ensembles
3. Run ensemble experiment
4. Create a new case
5. Load results manually into the new case
6. Run analysis with the new case as source case

Expectation:
Would expect a validation that said: No parameters for case: my_case (or something to that effect). The error message we get is not too bad, but it is an unknown exception, which is not great:
```Unable to run analysis for case 'my_case'.
The following error occurred: Unknown exception occurred with error: "No dataset 'COEFFS' in storage for realization 0"

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #6736 

**Approach**
Remove the state map from local_ensemble.py. This is replaced  by checking the state based on the actual files present in the ensemble directory on disk. We also store any errors with a possible error message on disk in the ensemble.

## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tests/unit_tests/test_load_forward_model.py [hunk 1]
```diff
@@ -13,7 +13,8 @@
 from ert.enkf_main import create_run_path, ensemble_context
 from ert.libres_facade import LibresFacade
 from ert.storage import open_storage
-from ert.storage.realization_storage_state import RealizationStorageState
+
+# from ert.storage.realization_storage_state import RealizationStorageState
```
Comment (dafeda) [inline tests/unit_tests/test_load_forward_model.py:17]:
```suggestion
```

File: tests/unit_tests/storage/test_local_storage.py [hunk 2]
```diff
@@ -38,6 +43,69 @@ def test_create_ensemble(tmp_path):
...
+            print(error.json(), file=f)
+
+        ensemble_2 = storage.create_ensemble(
+            experiment=experiment_id,
+            name="foo2",
+            ensemble_size=42,
+            prior_ensemble=ensemble_1,
+        )
+        assert _cases(storage) == ["foo1", "foo2"]
+        assert (
```
Comment (dafeda) [inline tests/unit_tests/storage/test_local_storage.py:103]: Both this test and `test_create_ensemble_from_prior` end with the same assert. Is it enough to have one of them?

File: tests/unit_tests/simulator/test_simulation_context.py [hunk 3]
```diff
@@ -62,11 +61,7 @@ def test_simulation_context(setup_case, storage, monkeypatch, try_queue_and_sche
             assert even_ctx.didRealizationSucceed(iens)
             assert not even_ctx.didRealizationFail(iens)
             assert even_ctx.isRealizationFinished(iens)
-
-            assert even_half.state_map[iens] == RealizationStorageState.HAS_DATA
```
Comment (dafeda) [inline tests/unit_tests/simulator/test_simulation_context.py:66]: Add a different assert here or OK to remove?

File: tests/unit_tests/simulator/test_simulation_context.py [hunk 4]
```diff
@@ -62,11 +61,7 @@ def test_simulation_context(setup_case, storage, monkeypatch, try_queue_and_sche
...
         else:
             assert odd_ctx.didRealizationSucceed(iens)
             assert not odd_ctx.didRealizationFail(iens)
             assert odd_ctx.isRealizationFinished(iens)
-
-            assert odd_half.state_map[iens] == RealizationStorageState.HAS_DATA
```
Comment (dafeda) [inline tests/unit_tests/simulator/test_simulation_context.py:72]: Add a different assert here or OK to remove?

File: tests/unit_tests/gui/test_full_manual_update_workflow.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import shutil
 
 import numpy as np
+import pytest
```
Comment (dafeda) [inline tests/unit_tests/gui/test_full_manual_update_workflow.py:4]:
```suggestion
```

File: src/ert/libres_facade.py [hunk 6]
```diff
@@ -173,17 +185,46 @@ def load_from_forward_model(
...
         return nr_loaded
 
+    def _load_from_run_path(
+        self,
+        ensemble_size: int,
+        run_args: List[RunArg],
+        active_realizations: List[bool],
+    ) -> int:
+        """Returns the number of loaded realizations"""
+        pool = ThreadPool(processes=8)
```
Comment (dafeda) [inline src/ert/libres_facade.py:206]: Just curious, why 8 threads?

**Review by dafeda — APPROVED** [review]
Just a few minor comments.
Great work 👍

File: src/ert/libres_facade.py (hunk 6 above)
Comment (frode-aarstad) [reply src/ert/libres_facade.py:206]: I just moved this code so I dont know

File: tests/unit_tests/simulator/test_simulation_context.py (hunk 3 above)
Comment (frode-aarstad) [reply tests/unit_tests/simulator/test_simulation_context.py:66]: Should be safe to remove as this test only simulates and not run any analysis thus has no data

File: tests/unit_tests/simulator/test_simulation_context.py (hunk 4 above)
Comment (frode-aarstad) [reply tests/unit_tests/simulator/test_simulation_context.py:72]: As above

File: tests/unit_tests/storage/test_local_storage.py (hunk 2 above)
Comment (frode-aarstad) [reply tests/unit_tests/storage/test_local_storage.py:77]: Good catch. Not really needed with two tests here

**Pull request merged**